### PR TITLE
Move rubyfmt to its own non-blocking job

### PR DESCRIPTION
**Pull request by elliottt**
The `test-static-sanitized.sh` job is made quite a bit longer by the requirement that we build `rubyfmt` to run the cli tests. This PR moves the rubyfmt tests to their own job, and removes `--config=rubyfmt` from the buildfarm configuration.

### Motivation

Improving build times.

### Test plan

See included automated tests.

### CODE REVIEW

File: .buildkite/pipeline.yaml
Comment (jez) [inline .buildkite/pipeline.yaml:46]: Curious what you think about moving this to be a job that happens down at the bottom of the file (next to the other non-blocking job)?

Right now as written this is still a blocking job.

File: .buildkite/test-sorbet-rubyfmt.sh
Comment (jez) [inline .buildkite/test-sorbet-rubyfmt.sh:3]: I think that the buildkite yaml pipeline syntax allows setting environment variables. Do you think we could set an environment variable and read that inside the existing `test-static-sanitized.sh` file so that we don't have to duplicate so much code?

File: .buildkite/pipeline.yaml
Comment (elliottt) [reply .buildkite/pipeline.yaml:46]: Ah, that was the intention originally. I had thought that always returning `0` would make it so that the job could fail, but that the failure wouldn't hold up subsequent jobs.

File: .buildkite/test-sorbet-rubyfmt.sh
Comment (elliottt) [reply .buildkite/test-sorbet-rubyfmt.sh:3]: Seems reasonable to me!

File: .buildkite/pipeline.yaml
Comment (jez) [reply .buildkite/pipeline.yaml:46]: Ah yeah that would work, but it would still potentially spend an instance running these tests before you could click the merge button.

**Review by jez — APPROVED** [review]
🙏

**Pull request merged**
